<div dir="rtl" style="text-align:right">
<h1 id="מחברת-לשיעור-התרגול-1">מחברת לשיעור התרגול — 1</h1>
<p dir="rtl" style="text-align:right">נקרא את ההסבר, נחזה תוצאה, נריץ את התא הבא ונפרש את הפלט לפני שנמשיך. נשתמש ב־<strong>Restart Kernel and Run All</strong> כדי לוודא שהעבודה אינה תלויה בהרצה ישנה. ההסברים, הגזירות המתמטיות וההנחות הפיסיקליות מהפרק נשמרים סביב התאים.</p>
</div>

<div dir="rtl" style="text-align:right">
<h2 id="הכנה-מסופקת-למחברת">הכנה מסופקת למחברת</h2>
<p dir="rtl" style="text-align:right">נבחר ליבה <strong>Julia 1.12</strong> ונפתח את המחברת מתוך תיקיית הפרק שלה. ההכנה מאתרת את סביבת הקורס המתועדת ויוצרת תיקייה נפרדת לאיורים; היא אינה מתקינה חבילות. אלה פקודות תשתית מסופקות, ולא חומר חדש להערכה.</p>
</div>

In [1]:
import Pkg
course_root = abspath("../..")
if !isfile(joinpath(course_root, "Project.toml"))
    error("Start this notebook from its week folder, e.g. week-1/notebooks")
end
Pkg.activate(course_root; io=devnull)
output_dir = get(ENV, "COURSE_NOTEBOOK_OUTPUT", joinpath(pwd(), "generated", "recitation-he"))
mkpath(output_dir)
println("Course environment active; figure output directory ready.")

Course environment active; figure output directory ready.


<div dir="rtl" style="text-align:right">
<h1 id="ביטויים-ראשונים-ב-julia-והתאמת-קפיץ">ביטויים ראשונים ב-<code dir="ltr">Julia</code> והתאמת קפיץ</h1>
<p dir="rtl" style="text-align:right"><em>סטודיו שבוע 1: קריאת טבלת קפיץ נתונה, התאמת שיפוע והתחלת תרגיל הליבה</em></p>
<h2 id="דף-עזר-במסך-אחד">דף עזר במסך אחד</h2>
<h4 id="מה-אנו-כבר-יודעים-לעשות">מה אנו כבר יודעים לעשות</h4>
<p dir="rtl" style="text-align:right">אנו יכולים להריץ תא ב-<code dir="ltr">Julia</code>, לחשב ביטויים סקלריים, להציב ולשנות ערכים, להדפיס תוצאה בעלת תווית ולהבחין בין שגיאת תחביר, שגיאת זמן ריצה ושגיאה לוגית. אנו גם יכולים לקרוא <code dir="ltr">CSV</code> נתון אל <code dir="ltr">DataFrame</code>, לבחור עמודות בשמן ולהעתיק ולפרש מתכון התאמה נתון בלי לטעון לשליטה במערכים או בהתאמה.</p>
<h4 id="רעיונות-מרכזיים-ועזר-תחביר">רעיונות מרכזיים ועזר תחביר</h4>
<p dir="rtl" style="text-align:right"><em>ביטוי</em> מפיק ערך. <em>משתנה</em> נותן שם לערך. ב-<code dir="ltr">name = expression</code>, <code dir="ltr">Julia</code> מחשבת את צד ימין ושומרת אותו בשם שמשמאל. נשתמש ב-<code dir="ltr">println("label", value)</code> לפלט קריא. שגיאת תחביר אינה ניתנת לניתוח; שגיאת זמן ריצה מתרחשת בזמן החישוב; שגיאה לוגית רצה אך עונה על השאלה הלא נכונה.</p>
<h4 id="הכלים-של-היום">הכלים של היום</h4>
<p dir="rtl" style="text-align:right">ערכים סקלריים, <code dir="ltr">+</code>, <code dir="ltr">-</code>, <code dir="ltr">*</code>, <code dir="ltr">/</code>, הצבה ב-<code dir="ltr">=</code>, <code dir="ltr">println</code>, <code dir="ltr">CSV.read</code>, <code dir="ltr">DataFrame</code> ובחירת עמודה בשם, הערות יחידה ובדיקת המקרה הידוע <span dir="ltr" class="math inline">\(V=0\Rightarrow I=0\)</span>. מתכון שיעורי הבית הנתון מכיל גם <code dir="ltr">length</code>, <code dir="ltr">sum</code> ואופרטורים מנוקדים; אלה אינם מיומנויות עצמאיות היום.</p>
<h4 id="בדיקה-לפני-שמתחילים">בדיקה לפני שמתחילים</h4>
<p dir="rtl" style="text-align:right">בלי להריץ קוד, חזו את הערך המספרי ואת היחידה שלו עבור <code dir="ltr">resistance</code> לאחר <code dir="ltr">voltage = 2.0</code>, <code dir="ltr">current = 0.4</code> ו-<code dir="ltr">resistance = voltage / current</code>.</p>
<p dir="rtl" style="text-align:right">להסברים מלאים חזרו לפרק שבוע 1 בחוברת ההרצאות, בסעיפים <em>היכן רואים תוצאה?</em>, <em><code dir="ltr">Julia</code> כמחשבון</em>, <em>שמות לערכים לפני הפיסיקה</em> ו-<em>תהליך מודרך ממדידה למודל</em>, ובהן התיבות על מינימום ריבועים, שאריות, יחידות ושגיאת מודל.</p>
<h2 id="הסטודיו-המונחה-17-דקות">הסטודיו המונחה: 17 דקות</h2>
<h4 id="תיבת-פיסיקה-לסטודנטים-קפיץ">תיבת פיסיקה לסטודנטים: קפיץ</h4>
<p dir="rtl" style="text-align:right">לקפיץ הנמתח מרחק מתון בתחום האלסטי מתקיים בגודל <span dir="ltr" class="math inline">\(F=kx\)</span>. ההתארכות <span dir="ltr" class="math inline">\(x\)</span> נמדדת מן האורך הלא-עמוס במטרים, הכוח <span dir="ltr" class="math inline">\(F\)</span> בניוטון, והשיפוע <span dir="ltr" class="math inline">\(k\)</span> הוא קבוע הקפיץ בניוטון למטר. נבחר התארכות וכוח חיוביים. במקרה הידוע <span dir="ltr" class="math inline">\(x=0\)</span> מתקבל <span dir="ltr" class="math inline">\(F=0\)</span>. עומס קדם, היסט, היסטרזיס, עיוות פלסטי ויציאה מן התחום האלסטי עלולים לפסול ישר.</p>
<h3 id="דקות-קריאת-csv-ו-dataframe">0–5 דקות: קריאת <code dir="ltr">CSV</code> ו-<code dir="ltr">DataFrame</code></h3>
<p dir="rtl" style="text-align:right">נפתח את קובץ הנתונים של הקורס <code dir="ltr">assets/data/spring_force.csv</code>. קובץ <code dir="ltr">CSV</code> הוא טבלת טקסט עם ערכים המופרדים בפסיקים ושמות עמודות בשורה הראשונה. נקרא אותו במפורש אל <code dir="ltr">DataFrame</code>: טבלה שבה עמודות בעלות שם נשארות מיושרות שורה-מול-שורה.</p>
</div>

<div dir="rtl" style="text-align:right">
<p dir="rtl" style="text-align:right"><strong>לפני ההרצה:</strong> נחזה את הערך, הטקסט או הגרף המתוארים לעיל. זו הדוגמה המתאימה מהפרק הכתוב.</p>
</div>

In [2]:
using CSV, DataFrames
spring = CSV.read(joinpath(course_root, "assets", "data", "spring_force.csv"), DataFrame)
x = spring.extension_m
y = spring.force_N
println(length(x), " measurements")

6 measurements

<div dir="rtl" style="text-align:right">
<p dir="rtl" style="text-align:right"><strong>בדיקה:</strong> נשווה את התוצאה להסבר שבהמשך. תא מציג אוטומטית רק את הביטוי האחרון; <code dir="ltr">println</code> מדפיסה טקסט במפורש. לאחר שינוי קלט נריץ שוב את התאים התלויים בו.</p>
</div>

<div dir="rtl" style="text-align:right">
<p dir="rtl" style="text-align:right">פלט הייחוס בפרק הכתוב, להשוואה לפלט שהתקבל בפועל:</p>
<pre dir="ltr" style="text-align:left" class="text"><code dir="ltr">6 measurements</code></pre>
</div>

<div dir="rtl" style="text-align:right">
<p dir="rtl" style="text-align:right">נריץ את דוגמת קריאת הקובץ מתיקיית הנתונים של הקורס, <code dir="ltr">assets/data</code>. הווקטורים <code dir="ltr">x</code> ו-<code dir="ltr">y</code> מכילים שש מדידות תואמות. נחזה <span dir="ltr" class="math inline">\(F=0\)</span> כאשר <span dir="ltr" class="math inline">\(x=0\)</span> ומגמה עולה וישרה. ניקוי טבלאות עשיר יותר יגיע בהמשך.</p>
<h3 id="דקות-התאמה-מודרכת-אחת">5–13 דקות: התאמה מודרכת אחת</h3>
<p dir="rtl" style="text-align:right">המחברת משתמשת במתכון הסכומים הממורכזים של ההרצאה. אנו קוראים את תוצאתו ולא מתכננים אותו מחדש.</p>
</div>

<div dir="rtl" style="text-align:right">
<p dir="rtl" style="text-align:right"><strong>לפני ההרצה:</strong> נחזה את הערך, הטקסט או הגרף המתוארים לעיל. זו הדוגמה המתאימה מהפרק הכתוב.</p>
</div>

In [3]:
x = [0.00, 0.02, 0.04, 0.06, 0.08, 0.10]   # m
y = [0.01, 0.39, 0.81, 1.18, 1.61, 1.99]    # N
n = length(x)
xbar = sum(x) / n; ybar = sum(y) / n
slope = sum((x .- xbar) .* (y .- ybar)) / sum((x .- xbar).^2)
intercept = ybar - slope * xbar
println("slope = $(round(slope; digits=1)) N/m")
println("intercept = $(round(intercept; digits=4)) N")

slope = 19.9 N/m


intercept = 0.0033 N


<div dir="rtl" style="text-align:right">
<p dir="rtl" style="text-align:right"><strong>בדיקה:</strong> נשווה את התוצאה להסבר שבהמשך. תא מציג אוטומטית רק את הביטוי האחרון; <code dir="ltr">println</code> מדפיסה טקסט במפורש. לאחר שינוי קלט נריץ שוב את התאים התלויים בו.</p>
</div>

<div dir="rtl" style="text-align:right">
<p dir="rtl" style="text-align:right">פלט הייחוס בפרק הכתוב, להשוואה לפלט שהתקבל בפועל:</p>
<pre dir="ltr" style="text-align:left" class="text"><code dir="ltr">slope = 19.9 N/m
intercept = 0.0033 N</code></pre>
</div>

<div dir="rtl" style="text-align:right">
<p dir="rtl" style="text-align:right">יחידות השיפוע הן ניוטון למטר, ולכן הוא מעריך <span dir="ltr" class="math inline">\(k\approx19.9\ \mathrm{N/m}\)</span>. חיתוך קטן הוא בדיקת היסט, לא חוק חדש.</p>
<p dir="rtl" style="text-align:right">אם נשנה את <code dir="ltr">voltage</code> ל-<code dir="ltr">3.0</code> בלי להריץ מחדש את ההשמה הסופית, <code dir="ltr">resistance</code> שומר על ערכו הישן. הצבה רושמת תוצאה כאשר השורה רצה.</p>
<h3 id="דקות-בדיקה-ופירוש">13–17 דקות: בדיקה ופירוש</h3>
<p dir="rtl" style="text-align:right">נבדוק אם השאריות משני צדי האפס. נשנה כוח נתון אחד ונבצע התאמה מחדש: גם השיפוע והחיתוך עשויים להשתנות, ולכן נחשב מחדש את כל התחזיות ואת כל השאריות לפני שנחליט מה השתנה. ישר עשוי להועיל בתחום הנתון בלי להוכיח שהקפיץ לינארי בכל התארכות.</p>
<h4 id="נקודת-בדיקה">נקודת בדיקה</h4>
<p dir="rtl" style="text-align:right">מה יחידות השיפוע? מדוע <span dir="ltr" class="math inline">\(F=0\)</span> ב-<span dir="ltr" class="math inline">\(x=0\)</span> היא בדיקה שימושית?</p>
<h3 id="דקות-העברה-לשיעורי-הבית">17–20 דקות: העברה לשיעורי הבית</h3>
<p dir="rtl" style="text-align:right">נפתח את הדף הנפרד <em>תרגיל בית 1: התאמת נפילה חופשית</em>. נאתר את <code dir="ltr">free_fall.csv</code> בתיקיית הנתונים של הקורס, ואז נזהה את <code dir="ltr">t_s</code>, את <code dir="ltr">y_m</code>, את התחזיות ואת השאריות.</p>
<h4 id="לאחר-דקה-17">לאחר דקה 17</h4>
<p dir="rtl" style="text-align:right">החלק המונחה של התרגול הסתיים. כל הזמן הנותר מוקדש לעבודה נתמכת על שיעורי בית הליבה הנפרדים בלבד. לא נוסיף פעילות תרגול נוספת ולא נפתח את הבונוס לפני שהליבה הושלמה באופן עצמאי.</p>
<h2 id="תרגול-מדידת-קבוע-קפיץ">תרגול: מדידת קבוע קפיץ</h2>
<p dir="rtl" style="text-align:right">נעביר את אותו ניתוח לקפיץ. זוגות המדידות הם התארכות <span dir="ltr" class="math inline">\(x\)</span> במטרים וכוח מופעל <span dir="ltr" class="math inline">\(F\)</span> בניוטון; נתאים את המודל <span dir="ltr" class="math inline">\(F=kx+F_0\)</span>. כאן, בניגוד לגרף שבו ההתארכות בציר האנכי, השיפוע הוא ישירות קבוע הקפיץ <span dir="ltr" class="math inline">\(k\)</span> ביחידות ניוטון למטר. נצפה מראש לשיפוע חיובי ונבדוק שלחותך יש יחידות כוח.</p>
<p dir="rtl" style="text-align:right">לפני המפגש נפתח את מחברת התרגול לשבוע 1 ונאתר את הנתונים ואת תאי ההתאמה המסופקים. בעשרים הדקות הראשונות נבחן את הצירים, נעקוב אחר התאמה אחת ונפרש את השיפוע, ההיסט ושארית אחת. לאחר מכן נתחיל את מטלת הנפילה החופשית בעזרת צוות ההוראה. נקודת בדיקה מועילה היא להסביר מדוע החלפת הצירים משנה את יחידות השיפוע ואת משמעותו; בנתונים רועשים, התאמה מחודשת בצירים ההפוכים אינה בהכרח הופכי השיפוע הקודם.</p>
<h2 id="מטלת-החובה-הסקת-תאוצת-הכובד-ממדידות-נפילה-חופשית">מטלת החובה: הסקת תאוצת הכובד ממדידות נפילה חופשית</h2>
<p dir="rtl" style="text-align:right">נבדוק אם מרחקי הנפילה מתאימים לשחרור ממנוחה בכבידה אחידה בקירוב. נגדיר מרחק חיובי כלפי מטה, ולכן <span dir="ltr" class="math inline">\(y(t)=\tfrac12gt^2+b\)</span>, כאשר <span dir="ltr" class="math inline">\(g&gt;0\)</span> ביחידות מטר לשנייה בריבוע ו-<span dir="ltr" class="math inline">\(b\)</span> במטרים. הפעולה המסופקת <code dir="ltr">t_s .^ 2</code> הופכת את ריבוע הזמן לקואורדינטה האופקית. נתאים <span dir="ltr" class="math inline">\(y=at^2+b\)</span> ונפרש <span dir="ltr" class="math inline">\(g=2a\)</span>: הגורם שתיים נובע ממשוואת התנועה ולא ממוסכמה של התוכנה.</p>
<p dir="rtl" style="text-align:right">נעבוד בשלבים:</p>
<ol type="1">
<li dir="rtl" style="text-align:right"><p dir="rtl" style="text-align:right">נבחן את <code dir="ltr">free_fall.csv</code> המסופק: נזהה את עמודות הזמן והמרחק, את יחידותיהן ואת מספר המדידות.</p></li>
<li dir="rtl" style="text-align:right"><p dir="rtl" style="text-align:right">נחזה את צורת הגרפים של <span dir="ltr" class="math inline">\(y\)</span> מול <span dir="ltr" class="math inline">\(t\)</span> ושל <span dir="ltr" class="math inline">\(y\)</span> מול <span dir="ltr" class="math inline">\(t^2\)</span> לפני ההתאמה. ניישם את המתכון ונשמור את התחזיות ואת השאריות, מדידה פחות תחזית.</p></li>
<li dir="rtl" style="text-align:right"><p dir="rtl" style="text-align:right">נכין איור אחד: הנקודות הנמדדות כסמנים עם המודל המותאם מעליהן, וכל ציר מסומן בכמותו וביחידותיו.</p></li>
<li dir="rtl" style="text-align:right"><p dir="rtl" style="text-align:right">נשמור את האיור כ-<code dir="ltr">PNG</code> או <code dir="ltr">PDF</code> ונדווח על <span dir="ltr" class="math inline">\(g\)</span> ועל ההיסט. נסביר אם השאריות תומכות במודל בתחום המדידות ונציין הנחה שההתאמה לבדה אינה מוכיחה.</p></li>
<li dir="rtl" style="text-align:right"><p dir="rtl" style="text-align:right">נגיש את הסקריפט ואת האיור בעמוד ה-<code dir="ltr">Moodle</code> של הקורס, כפי שמפורט בדף המטלה.</p></li>
</ol>
<p dir="rtl" style="text-align:right">נגיש סקריפט שלם יחד עם האיור; בשבוע 1 הגדרת פונקציה אינה חלק מהעבודה. המטלה הראשונה נבדקת ידנית ב-<code dir="ltr">Moodle</code>, עבור/לא עבור. כדאי להשאיר את דף המטלה לצידנו, מפני שבו מפורטים העמודות הנדרשות, האיור והגדלים שיש לדווח. עיגול התוצאה לצורך הצגה אינו צריך לעגל את הערכים המשמשים לחישוב.</p>
<h2 id="בונוס-לבחירה-גרף-שמעביר-תוצאה-מדעית">בונוס לבחירה: גרף שמעביר תוצאה מדעית</h2>
<p dir="rtl" style="text-align:right">הבונוס הוא רשות, אינו תנאי להמשך ואינו נבחן. השלמתו מזכה ב<strong>שליש נקודה בציון הסופי</strong>.</p>
<p dir="rtl" style="text-align:right">לאחר קריאה עצמאית של 10–15 דקות נשפר גרף מינימלי מסופק של חוק אוהם. כאן המתח בציר האופקי והזרם בציר האנכי, ולכן שיפוע המודל הוא מוליכות בסימנס ולא התנגדות באוהם. נזהה את סדרת הזרם הנמדד ואת סדרת הזרם המותאם לפני שינוי העיצוב.</p>
<p dir="rtl" style="text-align:right">נוסיף את הכותרת, תוויות הצירים והיחידות, המקרא, סמני המדידות, קו המודל, המסגרת וגודל האיור שנקבעו, ונייצא <code dir="ltr">PNG</code> ו-<code dir="ltr">PDF</code>. קטע החישוב מספק סדרות מאומתות ותואמות; התמונות נבדקות בנפרד לפי הרשימה המפורשת. נוודא ששינוי תווית או סגנון קו אינו משנה מדידה. דף הבונוס מציין את הטקסטים ושמות הקבצים המדויקים, כך שהבדיקה עוסקת בשלמות מדעית ולא בטעם אישי או בהשוואת פיקסלים.</p>
<h2 id="להעמקה-עקומות-ליסאזו">להעמקה: עקומות ליסאז׳ו</h2>
<p dir="rtl" style="text-align:right">בחקירה זו, שהיא רשות ואינה נבחנת, נשתמש במתכון שרטוט מסופק עבור <span dir="ltr" class="math display">\[x(t)=A\cos(at+\delta),\qquad y(t)=B\sin(bt).\]</span> הזמן <span dir="ltr" class="math inline">\(t\)</span> בשניות, התדירויות הזוויתיות <span dir="ltr" class="math inline">\(a,b\)</span> ברדיאנים לשנייה, הפאזה <span dir="ltr" class="math inline">\(\delta\)</span> ברדיאנים, ו-<span dir="ltr" class="math inline">\(A,B\)</span> הן משרעות הקואורדינטות. העקומה מציגה שתי תנועות סינוסיות בלתי תלויות יחד; אין זה מסלול תחת כוח כבידה מרכזי.</p>
<p dir="rtl" style="text-align:right">נקבע משרעות ונשווה יחסי תדירות <span dir="ltr" class="math inline">\(1:1\)</span>, <span dir="ltr" class="math inline">\(2:1\)</span> ו-<span dir="ltr" class="math inline">\(3:2\)</span>. עבור יחס אחד נשנה את הפאזה מ-<span dir="ltr" class="math inline">\(0\)</span> ל-<span dir="ltr" class="math inline">\(\pi/2\)</span>. נשמור שני גרפים מנוגדים וננסח השערה על סגירות ועל סימטריה. נבדוק אם הארכת תחום הזמן משנה עקומה שנראתה פתוחה: חלון שרטוט סופי אינו הוכחה לאי-סגירות. הפעילות אינה מוסיפה תחביר חובה והיא נפרדת מהבונוס המזכה בנקודות.</p>
</div>

<div dir="rtl" style="text-align:right">
<h2 id="דוגמאות-קטנות-ובדיקות-המוזכרות-בפרק">דוגמאות קטנות ובדיקות המוזכרות בפרק</h2>
<p dir="rtl" style="text-align:right">התאים הבאים מאפשרים להריץ דוגמאות קצרות ובדיקות שהוסברו בטקסט. הם אינם מוסיפים פתרונות לשיעורי הבית.</p>
</div>

<div dir="rtl" style="text-align:right">
<p dir="rtl" style="text-align:right">בדיקת פתיחה: מתח חלקי זרם נותן התנגדות. שינוי המתח אינו מחשב מחדש התנגדות שכבר נשמרה.</p>
</div>

In [4]:
voltage = 2.0      # V
current = 0.4      # A
resistance = voltage / current   # ohm
println(resistance)
voltage = 3.0      # V
println(resistance)

5.0
5.0


<div dir="rtl" style="text-align:right">
<h2 id="סיכום-ופרשנות">סיכום ופרשנות</h2>
<p dir="rtl" style="text-align:right">נסביר תוצאה אחת במילים, נציין יחידות או הנחות כשצריך ונזהה בדיקה שעשויה לחשוף טעות. תיאורי התרגול, מטלת החובה, הבונוס וההעמקה שלעיל מכוונים להמשך. רקע לבחירה אינו הופך לתנאי למבחן.</p>
</div>